api/src/services/bcgw-layer-service.ts

```diff
@@ -0,0 +1,460 @@
+    // Array of all matching region details for the feature
+    let response: RegionDetails[] = [];
+
+    const postgisService = new PostgisService(connection);
+    // Convert the feature geometry to WKT format
+    const result = await postgisService.getGeoJsonGeometryAsWkt(feature.geometry, Srid3005);
+    const geometryWKTString = result.geometry;
+
+    // Attempt to detect if the feature is a known BCGW feature
+    const regionDetails = this.findRegionDetails(feature);
+
+    if (!regionDetails) {
+      // Feature is not a known BCGW feature
+      // Fetch region details for the feature from ALL available layers
+      response = response.concat(
+        await this.getAllRegionDetailsForWktString(geometryWKTString, [
+          BcgwEnvRegionsLayer,
+          BcgwNrmRegionsLayer
+          // BcgwParksAndEcoreservesLayer,
+          // BcgwWildlifeManagementUnitsLayer
+        ])
+      );
+    } else {
+      // Feature is a known BCGW feature, fetch any additional mapped region details, and add to the overall response
+      const mappedRegionDetails = this.getMappedRegionDetails([regionDetails]);
+      response = response.concat(mappedRegionDetails);
+      // Fetch region details for the feature, excluding the layer whose details were already added above
+      // Why? We want to avoid querying a layer using a feature from that same layer because the actual results will not
+      // be consistent with the expected results. (Ex: LayerA + Feature1 should return Feature1, but will often return
+      // Feature1 + Feature2 + Feature3 due to the layer features containing overlapping coordinates, when ideally they
+      // should not).
+      const layersToProcess = [
+        BcgwEnvRegionsLayer,
+        BcgwNrmRegionsLayer
+        //   BcgwParksAndEcoreservesLayer,
+        //   BcgwWildlifeManagementUnitsLayer
+      ].filter(
+        (layerToProcess) =>
+          !mappedRegionDetails.map((mappedRegionDetail) => mappedRegionDetail.sourceLayer).includes(layerToProcess)
+      );
+
+      response = response.concat(await this.getAllRegionDetailsForWktString(geometryWKTString, layersToProcess));
+    }
+
+    return response;
+  }
+
+  /**
+   * Given a geometry WKT string and array of layers to process, return an array of all matching region details for the
+   * specified layers.
+   *
+   * @param {string} geometryWktString a geometry string in Well-Known Text format
+   * @param {string[]} layersToProcess an array of supported layers to query against
+   * @return {*}
+   * @memberof BcgwLayerService
+   */
+  async getAllRegionDetailsForWktString(geometryWktString: string, layersToProcess: string[]) {
+    let response: RegionDetails[] = [];
+
+    for (const layerToProcess of layersToProcess) {
+      switch (layerToProcess) {
+        case BcgwEnvRegionsLayer:
+          response = response.concat(await this.getEnvRegionDetails(geometryWktString));
```

Comment:
Could we improve readibility by using `response.push` instead of `response = response.concat`?

Reply:
Nevermind, I misunderstood what was happening 😅